### ISSUE TITLE
Call Host function

### ISSUE THREAD

**Issue by Tails** [second-state/wasmedge-quickjs#86]
As I understand it, Rust's ```main()``` function is compiled to WASM. Is it possible to call a non-WASM host function so that Javascript could actually call to the host outside of the runtime?

My case is that I am drawing on a canvas from my JS code, but no canvas component exists there (and Skia cannot be compiled to wasm-wasi), so I have to pass the draw commands and draw _outside_ of the runtime.

**Comment by juntao** [comment]
Yes, it is a two step process. First, you can create a plugin for host functions. These host functions can be accessed from inside WasmEdge as Rust API. 

https://github.com/WasmEdge/WasmEdge/tree/master/examples/plugin/

Then in QuickJS, you can create a JS API for the host function API. 

https://wasmedge.org/book/en/write_wasm/js/rust.html
